File: Java/src/com/algoexpert/twonumbersum/twoNumberSumUsingSort.py
# O(nlogn) time | O(1) space
def twoNumberSumUsingSort(array, targetSum):
    array.sort()
    left = 0
    right = len(array) - 1
    while left < right:
        currentSum = array[left] + array[right]
        if targetSum == currentSum:
            return [array[left], array[right]]
        elif targetSum > currentSum:
            left += 1
        else:
            right -= 1
    return []


new_array = [3, 5, -4, 8, 11, 1, -1, 6]
print(twoNumberSumUsingSort(new_array, 10))

File: Java/src/com/algoexpert/array/twonumbersum/TwoNumberSumUsingSort.java
package com.algoexpert.array.twonumbersum;

import java.util.Arrays;

public class TwoNumberSumUsingSort {
    // O(nlogn) time | O(1) space
    public static int[] twoNumberSumUsingLoop(int[] arr, int target) {
        Arrays.sort(arr);
        int left= 0;
        int right = arr.length-1;
        for(int i = 0; i < arr.length; i++) {
            int currentSum=arr[left] + arr[right];
            if(currentSum == target){
                return new int[]{arr[left], arr[right]};
            } else if(currentSum > target){
                right--;
            } else {
                left ++;
            }

        }
        return new int[0];
    }

    public static void main(String[] args) {
        int[] arr = new int[]{3, 5, -4, 8, 11, 1, -1, 6};
        int target = 10;
        int[] result = twoNumberSumUsingLoop(arr, target);
        System.out.println(result[0] + " + " + result[1] + " = " + target);
    }
}
